pkg/mqtrigger/kafka/main.go

```diff
@@ -0,0 +1,480 @@
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"hash"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	"github.com/xdg/scram"
+	"go.uber.org/zap"
+)
+
+type kafkaMetadata struct {
+	bootstrapServers []string
+
+	// auth
+	authMode kafkaAuthMode
+	username string
+	password string
+
+	// ssl
+	cert string
+	key  string
+	ca   string
+}
+
+type fissionMetadata struct {
+	// fission
+	topic         string
+	responseTopic string
+	errorTopic    string
+	functionURL   string
+	maxRetries    int
+	contentType   string
+	triggerName   string
+	consumerGroup string
```

Comment:
ConsumerGroup is specific to Kafka whereas the other attributes are neutral/common to all other MQ types too. I suggest:
1) Move CG to KafkaMetadata
2) Move fissionMetadata to a common file (As it will be common to all other integrations of future)

Reply:
Done